pkg/volume/util/volumehelper/volumehelper.go

```diff
@@ -52,6 +52,12 @@ func GetUniqueVolumeName(pluginName, volumeName string) api.UniqueVolumeName {
 	return api.UniqueVolumeName(fmt.Sprintf("%s/%s", pluginName, volumeName))
 }
 
+// GetUniqueVolumeNameForNonAttachableVolume returns the unique volume name
+// for a non-attachable volume.
+func GetUniqueVolumeNameForNonAttachableVolume(volumePlugin volume.VolumePlugin, podNamespace, podName, podSpecName string) api.UniqueVolumeName {
+	return api.UniqueVolumeName(fmt.Sprintf("%s/%s-%s-%s", volumePlugin.GetPluginName(), podNamespace, podName, podSpecName))
```

Comment:
@jingxu97 I'm changing this to use the pod UID instead of the namespace and name to aid you in #27970